### PR TITLE
Added the ability to get guild data from a guild's name

### PR DESCRIPTION
**Pull request by Mythbusters123**
This is probably coded very poorly, but it is based off what the current system is and it would be too hard to be able to parse between an ID or a name, so I went with this method. Should be the same output, just with a different input.

### CODE REVIEW

**Comment by Mythbusters123** [comment]
I just realized that I forgot to add an endpoint URL. I will fix this in the morning

File: .env_example [hunk 1]
```diff
@@ -1 +0,0 @@
-HYPIXEL_API_KEY=REPLACE_THIS
```
Comment (builder-247) [inline .env_example:1]: Why was this deleted?

File: package-lock.json [hunk 2]
```diff
@@ -3472,8 +3472,8 @@
       }
     },
     "hypixelconstants": {
-      "version": "github:slothpixel/hypixelconstants#2d92f504b4a8516a1dcc0a690653769569b024a9",
```
Comment (builder-247) [inline package-lock.json:3475]: Can you undo these changes?

**Comment by builder-247** [comment]
Thank you for your contribution!

You'll need to also add the route to graphql https://github.com/slothpixel/core/blob/master/routes/graphql.js

Also for the build to pass you should add tests or ignore the endpoint here https://github.com/slothpixel/core/blob/master/test/test.js#L111

**Comment by Mythbusters123** [comment]
I dont have any experience with GraphQL, however i will try my best. I will also undo the things you told me to. Commits should come soon.

File: routes/spec.js [hunk 3]
```diff
@@ -2311,4 +2467,4 @@ Consider supporting The Slothpixel Project on Patreon to help cover the hosting
   },
 };
 
-module.exports = spec;
+module.exports = { spec };
```
Comment (builder-247) [inline routes/spec.js:2470]: this breaks everything

**Comment by Mythbusters123** [comment]
why that break everything i have no idea. thanks for pointing that out :)

**Comment by builder-247** [comment]
Re-add .env_example and we are good to merge

**Comment by Mythbusters123** [comment]
Alright i will do that when i get back home. Thank you!

File: routes/spec.js [hunk 4]
```diff
@@ -847,6 +847,162 @@ Consider supporting The Slothpixel Project on Patreon to help cover the hosting
...
+                  },
+                },
+              },
+            },
+          },
+        },
+        route: () => '/guilds/name/:name',
+        func: async (request, response, callback) => {
+          try {
+            const guild = await getGuildFromName(request.params.player);
```
Comment (builder-247) [inline routes/spec.js:995]: You need to put `request.params.name` here, and also use the correct parameter in the graphql processor

Comment (Mythbusters123) [reply routes/spec.js:995]: sounds good!

Comment (builder-247) [reply routes/spec.js:995]: You should also add a graphql spec like here https://github.com/slothpixel/core/blob/master/routes/spec.graphql#L29

Comment (Mythbusters123) [reply routes/spec.js]: Alright!

File: routes/spec.graphql [hunk 5]
```diff
@@ -28,7 +28,7 @@ type Query {
   """
   guild(
     """
-    Username or non-dashed UUID of player. UUID should be used when feasible for better performance.
+    Username or non-dashed UUID of player. UUID should be used when feasible for better performance. Will use the Guild's Name if you use the syntax name/{guildName} to it.
```
Comment (builder-247) [inline routes/spec.graphql:31]: You should make a new method, named something like `guild_by_name`, then copy and paste the code from above and change `player_name` to `guild_name` + add an entry here

Comment (builder-247) [reply routes/spec.graphql:31]: Also sorry for not making a more thorough review the first time :P

Comment (Mythbusters123) [reply routes/spec.graphql:31]: Its fine this is my first time working with an API so i didnt know exactly how to make it lol.

**Review by builder-247 — APPROVED** [review]
